src/mine_loop.rs

```diff
@@ -433,23 +428,53 @@ pub(super) fn prepare_coinbase_transaction_stateless(
         .expect("Amount to composer must be larger than liquid amount to composer.");
 
     let owned = true;
-    let liquid_coinbase_output = TxOutput::offchain_native_currency(
+    let liquid_coinbase_output = TxOutput::native_currency(
         liquid_composer_amount,
         composer_parameters.sender_randomness(),
         composer_parameters.reward_address(),
+        composer_parameters.notification_policy().into(),
         owned,
     );
 
     // Set the time lock to 3 years (minimum) plus 30 minutes margin, since the
     // timestamp might be bumped by future merges.
-    let timelocked_coinbase_output = TxOutput::offchain_native_currency(
+    let timelocked_coinbase_output = TxOutput::native_currency(
         timelocked_composer_amount,
         composer_parameters.sender_randomness(),
         composer_parameters.reward_address(),
+        composer_parameters.notification_policy().into(),
         owned,
     )
     .with_time_lock(timestamp + MINING_REWARD_TIME_LOCK_PERIOD + Timestamp::minutes(30));
 
+    Ok([liquid_coinbase_output, timelocked_coinbase_output])
+}
+
+/// Compute `TransactionDetails` and a list of `TxOutput`s for a coinbase
+/// transaction.
+pub(super) fn prepare_coinbase_transaction_stateless(
+    latest_block: &Block,
+    composer_parameters: ComposerParameters,
+    timestamp: Timestamp,
+) -> Result<(TxOutputList, TransactionDetails)> {
+    let mutator_set_accumulator = latest_block.mutator_set_accumulator_after().clone();
+    let next_block_height: BlockHeight = latest_block.header().height.next();
+    info!("Creating coinbase for block of height {next_block_height}.");
+
+    let coinbase_amount = Block::block_subsidy(next_block_height);
+    let [liquid_coinbase_output, timelocked_coinbase_output] =
+        coinbase_outputs(coinbase_amount, composer_parameters, timestamp)?;
+    let total_composer_fee = liquid_coinbase_output.utxo().get_native_currency_amount()
+        + timelocked_coinbase_output
+            .utxo()
+            .get_native_currency_amount();
+    let total_guesser_fee = coinbase_amount.checked_sub(&total_composer_fee).unwrap();
```

Comment:
Is it part of the `coinbase_output()` contract that the total composer fee is less than or equal to the coinbase amount?